src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ActivatorUtilities.cs

```diff
@@ -508,16 +508,10 @@ private static bool IsService(IServiceProviderIsService serviceProviderIsService
 
         private static bool TryGetServiceKey(ParameterInfo parameterInfo, out object? key)
         {
-            if (parameterInfo.CustomAttributes != null)
+            foreach (var attribute in parameterInfo.GetCustomAttributes<FromKeyedServicesAttribute>(false))
```

Comment:
Since the return value of `GetCustomAttributes<FromKeyedServicesAttribute>(false)` is `Attribute[]`, setting up an enumeration to return on the first entry unconditionally seems more work. Would this be faster?

```csharp
Attribute[] attributes = parameterInfo.GetCustomAttributes<FromKeyedServicesAttribute>(false);
key = (attributes.IsEmpty) ? null : attributes[0];
return key != null;
```